be/src/olap/rowset/segment_v2/column_zone_map.cpp

```diff
@@ -79,10 +87,17 @@ Status ColumnZoneMapBuilder::flush() {
 
 void ColumnZoneMapBuilder::_reset_zone_map() {
     // we should allocate max varchar length and set to max for min value
-    Slice *min_slice = (Slice *)_zone_map.min_value;
-    min_slice->data = _max_string_value;
-    min_slice->size = OLAP_STRING_MAX_LENGTH;
-    _field->set_to_max(_zone_map.min_value);
+    if (_type_info->type() == OLAP_FIELD_TYPE_VARCHAR) {
+        Slice *min_slice = (Slice *)_zone_map.min_value;
+        min_slice->data = _max_varchar_value;
+        min_slice->size = OLAP_STRING_MAX_LENGTH;
+    } else if (_type_info->type() == OLAP_FIELD_TYPE_CHAR) {
+        Slice *min_value = (Slice *)_zone_map.min_value;
+        min_value->data = _max_char_value;
+        min_value->size = OLAP_CHAR_MAX_LENGTH;
```

Comment:
why set char type's max length to 255? I think it should be equal to OLAP_STRING_MAX_LENGTH.

Reply:
Fe defines max char and varchar length,see ScalarType;this is a temporary define,later I will use columns real length

Reply:
yes, for char type, it is a bit complex